### PR TITLE
Updating main branch for legacy libc++ support

### PR DESCRIPTION
**Pull request by lemire**
It appears that legacy libc++ requires special care.

### CODE REVIEW

File: include/simdjson/generic/numberparsing.h [hunk 1]
```diff
@@ -293,7 +293,10 @@ simdjson_really_inline bool compute_float_64(int64_t power, uint64_t i, bool neg
 static bool parse_float_fallback(const uint8_t *ptr, double *outDouble) {
   *outDouble = simdjson::internal::from_chars((const char *)ptr);
   // We do not accept infinite values.
-  if (!std::isfinite(*outDouble)) {
+  if ((*outDouble > 1.7976931348623157E+308) || (*outDouble < -1.7976931348623157E+308)) {
```
Comment (pauldreik) [inline include/simdjson/generic/numberparsing.h:296]: Would it work to use https://en.cppreference.com/w/cpp/types/numeric_limits/max instead? In my opinion, that would make the code self explaining.

File: src/to_chars.cpp [hunk 2]
```diff
@@ -914,8 +914,10 @@ format. Returns an iterator pointing past-the-end of the decimal representation.
 */
 char *to_chars(char *first, const char *last, double value) {
   static_cast<void>(last); // maybe unused - fix warning
-  // Use signbit(value) instead of (value < 0) since signbit works for -0.
-  if (std::signbit(value)) {
+  // Next line would be better but it mysteriously fail under legacy/old libc++ libraries.
+  // See https://github.com/simdjson/simdjson/issues/1286
+  // if (std::signbit(value)) {
+  if ((value < 0) || (value == -0)) {
```
Comment (pauldreik) [inline src/to_chars.cpp:920]: would value<=-0 work?

File: include/simdjson/generic/numberparsing.h (hunk 1 above)
Comment (lemire) [reply include/simdjson/generic/numberparsing.h]: You are correct. Modified.

File: src/to_chars.cpp (hunk 2 above)
Comment (lemire) [reply src/to_chars.cpp]: It should. Modified.

File: include/simdjson/generic/numberparsing.h (hunk 1 above)
Comment (lemire) [reply include/simdjson/generic/numberparsing.h]: @pauldreik Sadly, it appears to be unsupported under Visual Studio.

**Comment by pauldreik** [comment]
Hmm, is the visual studio problem related to that max and min sometimes are macros?

**Comment by lemire** [comment]
Merging.

**Comment by lemire** [comment]
> Hmm, is the visual studio problem related to that max and min sometimes are macros?

Could be. 

At some point, one needs to make progress. I have tried many things to check that the value is finite. This gets flat out ridiculous.

We can patch it up with a new PR.

**Pull request merged**